src/ws/cockpitauth.c

```diff
@@ -1164,13 +1164,22 @@ cockpit_session_launch (CockpitAuth *self,
       capture_stderr = cockpit_conf_bool ("WebService", "X-For-CockpitClient", FALSE);
       program_default = cockpit_ws_ssh_program;
     }
-  else
+  else if (type && (g_str_equal (type, "basic") ||
+                    g_str_equal (type, "negotiate") ||
+                    g_str_equal (type, "tls-cert")))
     {
       program_default = cockpit_ws_session_program;
     }
 
   command = type_option (section, "command", program_default);
 
+  if (!command)
+    {
+      g_set_error (error, COCKPIT_ERROR, COCKPIT_ERROR_AUTHENTICATION_FAILED,
+                   "Authentication disabled");
+      goto out;
```

Comment:
Just to make sure I fully understand this: Disabling an auth method with `action = none` already happens further up in line 1136. This code would only apply to a custom auth type that has no `command =`, so it's less about "disabled" and more about "misconfigured", right? (No need to change the string)

Reply:
> Just to make sure I fully understand this: Disabling an auth method with `action = none` already happens further up in line 1136. This code would only apply to a custom auth type that has no `command =`, so it's less about "disabled" and more about "misconfigured", right? (No need to change the string)

It's more about the case we discussed: random unexpected incoming Authorization headers.  The default "localhost" action will still be taken for those, but without a command configured, they will now fail.  That's slightly different than the "action" approach we discussed but it has the advantage of not breaking existing users of command= for strange auth types, including many in the unit tests.  If the user explicitly configured a command then they probably expect it to work.

Reply:
Ah right, due to the NULL program_default for unknown auth types, type_option() would return NULL for that as well. `</thinking_aloud>`. Thanks!